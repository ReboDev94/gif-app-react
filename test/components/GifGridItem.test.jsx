import { render, screen } from "@testing-library/react"

describe('pruebas en GifGridItem.jsx', () => {

    const img = {
        id: 'gk3R16JhLP8RUka2nD',
        title: 'Count Me In Rick And Morty GIF',
        url: 'https://media0.giphy.com/media/gk3R16JhLP8RUka2nD/giphy.gif?cid=687a5183yndu0bjk14vb9vwskbrg5q2a6pjcib7jjd85ey9i&rid=giphy.gif&ct=g'
    }

    test('hace match con el snapshot', () => {
        const { container } = render(<GifGridItem img={img} />)
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar con el url y el ALT indicado', () => {
        render(<GifGridItem img={img} />)
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(img.url)
        expect(alt).toBe(img.title)
    })
    test('debe haber un titulo en la imagen', () => {
        render(<GifGridItem img={img} />)
        expect(screen.getByText(img.title)).toBeTruthy()
    })
})
