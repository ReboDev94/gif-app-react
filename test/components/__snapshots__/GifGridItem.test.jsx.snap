// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem.jsx hace match con el snapshot 1`] = `
<div>
  <div
    class="w-full md:w-1/3 px-2 my-2"
  >
    <div
      class="rounded-md border"
    >
      <img
        alt="Count Me In Rick And Morty GIF"
        class="rounded-t-md h-72 object-cover mx-auto"
        src="https://media0.giphy.com/media/gk3R16JhLP8RUka2nD/giphy.gif?cid=687a5183yndu0bjk14vb9vwskbrg5q2a6pjcib7jjd85ey9i&rid=giphy.gif&ct=g"
      />
      <p
        class="p-2 text-ellipsis w-full"
      >
        Count Me In Rick And Morty GIF
      </p>
    </div>
  </div>
</div>
`;
